src/main/test/java/io/burt/jmespath/contrib/function/JmesPathRuntimeWithStringFunctionTest.java

```diff
@@ -0,0 +1,586 @@
+package io.burt.jmespath.contrib.function;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.regex.PatternSyntaxException;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import io.burt.jmespath.Adapter;
+import io.burt.jmespath.RuntimeConfiguration;
+import io.burt.jmespath.function.ArgumentTypeException;
+import io.burt.jmespath.function.FunctionRegistry;
+import io.burt.jmespath.parser.ParseException;
+
+public abstract class JmesPathRuntimeWithStringFunctionTest<T> extends JmesPathRuntimeTest<T> {
+  private FunctionRegistry functionRegistry = FunctionRegistry.defaultRegistry()
+          .extend(new ConcatFunction())
+          .extend(new LowerCaseFunction())
+          .extend(new MatchesFunction())
+          .extend(new NormalizeSpaceFunction())
+          .extend(new ReplaceFunction())
+          .extend(new SubstringAfterFunction())
+          .extend(new SubstringBeforeFunction())
+          .extend(new TokenizeFunction())
+          .extend(new TranslateFunction())
+          .extend(new UpperCaseFunction());
+
+  private Adapter<T> runtime = createRuntime(RuntimeConfiguration.builder()
+          .withFunctionRegistry(functionRegistry)
+          .build());
+
+  @Override
+  protected Adapter<T> runtime() { return runtime; }
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  private T dontCare, emptyObject, emptyList;
+
+  @Before
+  public void setUp() {
+    emptyObject = parse("{}");
+    emptyList = parse("[]");
+    dontCare = emptyObject;
```

Comment:
Why not simply use `emptyObject`? I think that would make the tests more clear that they run against a target that does not contain anything.